### PR TITLE
Warn on underspecified opaque types

### PR DESCRIPTION
**Pull request by fertapric**
Refers to #5800 

Performs the `erl_lint` check `{underspecified_opaque, {TypeName, Arity}}` in Elixir.

After reading the Erlang/OTP codebase, it seems an "underspecified opaque" is an opaque type of arity 0 with a spec of `term` or `any`. [Here is the implementation in the Erlang/OTP codebase](https://github.com/erlang/otp/blob/master/lib/stdlib/src/erl_lint.erl#L2796-L2798) and [the test suite](https://github.com/erlang/otp/blob/master/lib/stdlib/test/erl_lint_SUITE.erl#L2606-L2613).

### CODE REVIEW

File: lib/elixir/lib/kernel/typespec.ex [hunk 1]
```diff
@@ -283,6 +283,11 @@ defmodule Kernel.Typespec do
       compile_error(caller, message)
     end
 
+    if kind == :opaque and arity == 0 and underspecified?(spec) do
```
Comment (josevalim) [inline lib/elixir/lib/kernel/typespec.ex]: Maybe we should move the opaque and arity inside the `underspecified?` function?

Comment (fertapric) [reply lib/elixir/lib/kernel/typespec.ex]: I started with `underspecified?(kind, arity, spec)` but it felt weird that the message have the "opaque" hardcoded. 

But rethinking this part, maybe we can reword the message to include the kind? For example:

```elixir
if underspecified?(kind, arity, spec) do
  message = "@#{kind} type #{name}/#{arity} is underspecified and therefore meaningless"
  :elixir_errors.warn(caller.line, caller.file, message)
end
```

Comment (josevalim) [reply lib/elixir/lib/kernel/typespec.ex]: LGTM!

**Comment by whatyouhide** [comment]
Thanks @fertapric! 💟

**Pull request merged**